internal/pathutil/configpath.go

```diff
@@ -0,0 +1,159 @@
+package pathutil
+
+import (
+	"encoding/json"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+
+	"github.com/skycoin/skycoin/src/util/logging"
+)
+
+var log = logging.MustGetLogger("pathutil")
+
+// ConfigLocationType describes a config path's location type.
+type ConfigLocationType string
+
+const (
+	// WorkingDirLoc represents the default working directory location for a configuration file.
+	WorkingDirLoc = ConfigLocationType("WD")
+
+	// HomeLoc represents the default home folder location for a configuration file.
+	HomeLoc = ConfigLocationType("HOME")
+
+	// LocalLoc represents the default /usr/local location for a configuration file.
+	LocalLoc = ConfigLocationType("LOCAL")
+)
+
+// String implements fmt.Stringer for ConfigLocationType.
+func (t ConfigLocationType) String() string {
+	return string(t)
+}
+
+// Set implements pflag.Value for ConfigLocationType.
+func (t *ConfigLocationType) Set(s string) error {
+	*t = ConfigLocationType(s)
+	return nil
+}
+
+// Type implements pflag.Value for ConfigLocationType.
+func (t ConfigLocationType) Type() string {
+	return "pathutil.ConfigLocationType"
+}
+
+// AllConfigLocationTypes returns all valid config location types.
+func AllConfigLocationTypes() []ConfigLocationType {
+	return []ConfigLocationType{
+		WorkingDirLoc,
+		HomeLoc,
+		LocalLoc,
+	}
+}
+
+// ConfigPaths contains a map of configuration paths, based on ConfigLocationTypes.
+type ConfigPaths map[ConfigLocationType]string
+
+// String implements fmt.Stringer for ConfigPaths.
+func (dp ConfigPaths) String() string {
```

Comment:
Just one question, why did you chose to call this `dp`?

Reply:
Hi Ivan. I believe dp is "defaultPath". The original configpath.go code was actually written by Evan to manage the configpath for mainnet (https://github.com/skycoin/skywire/blob/mainnet/internal/pathutil/configpath.go).  I re-used Evan's code and incorporated a function that is specific for Skywire-Updater config. I hope that this doesn't cause any issues. Thank you.

Reply:
@atang152 What is the additional function you have implemented? Can we make use of the version in `skycoin/skywire` now?

Reply:
Hi Evan. In your configPath.go code, you created two funcs that returns default path for 1) skywire-node and 2) manager node. They are called "NodeDefaults()" and "ManagerDefaults" respectively. I added one extra func called UpdaterDefaults() that simply returns the default config paths specifically for Skywire-Updater. 

_Can we make use of the version in skycoin/skywire now?_ : It can be incorporated into the current version in skycoin/skywire but I don't think it will do anything since the current Skywire-Updater is independent and seperated from the Skywire-Updater. Please let me know if there is anything I should do. Thanks!

Reply:
@atang152 The changes that are required are as follows.
1. Import `github.com/skycoin/skywire/pkg/util/pathutil`, rather than copying the source to `github.com/skycoin/skywire-updater/...`.

    The command to achieve this (with env `GO111MODULE=on`) is: 
    ```golang
    go get github.com/skycoin/skywire/pkg/util/pathutil@mainnet
    ```
    Then you can import this in the code.

2. Please put `UpdaterDefaults()` in a more suitable location (in `/cmd/skywire-updater/root.go` is fine).

Reply:
@atang152 https://github.com/golang/go/wiki/modules

Hopefully this resource will help.

Reply:
> @atang152 The changes that are required are as follows.
> 
> 1. Import `github.com/skycoin/skywire/pkg/util/pathutil`, rather than copying the source to `github.com/skycoin/skywire-updater/...`.
>    The command to achieve this (with env `GO111MODULE=on`) is:
>    ```go
>    go get github.com/skycoin/skywire/pkg/util/pathutil@mainnet
>    ```
>    
>    
>    Then you can import this in the code.
> 2. Please put `UpdaterDefaults()` in a more suitable location (in `/cmd/skywire-updater/root.go` is fine).

@atang152 How is progress with this? Just referencing this here as I see the source comment is outdated.

https://github.com/golang/go/wiki/modules